src/dotnet/CommandResolution/LockFileTargetExtensions.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NuGet.Packaging.Core;
+using NuGet.ProjectModel;
+
+namespace Microsoft.DotNet.Cli.CommandResolution
+{
+    internal static class LockFileTargetExtensions
+    {
+        public static bool IsPortable(this LockFileTarget lockFileTarget)
+        {
+            return string.IsNullOrEmpty(lockFileTarget.RuntimeIdentifier) &&
+                lockFileTarget.GetPlatformLibrary() != null;
+        }
+
+        public static LockFileTargetLibrary GetPlatformLibrary(this LockFileTarget lockFileTarget)
+        {
+            // TODO: https://github.com/dotnet/sdk/issues/17 get this from the lock file
```

Comment:
I think we should open a new bug for this in the CLI repo.  What's the plan here?

The plan for understanding the 'platform' dependency for the project is to have an MSBuild property that gets passed in.  But will the 'platform' dependency for the current project always match all the tool's platform dependency?